src/wal.rs

```diff
@@ -1,10 +1,70 @@
 use super::Result;
+use crate::util::binary::{
```

Comment:
How about using a crate like `https://docs.rs/integer-encoding/2.1.1/integer_encoding/` instead of porting a go `binary`. /cc @BusyJay

Reply:
The latest patch seems to copy the implementation from a library written by me. Only three functions are necessary and they are quite simple. I suggest to avoid adding new dependencies and delete unused functions in the futures.

Reply:
I'll remove zig-zag functions and signed integer functions from `binary.rs`

Reply:
Done. Now we only retain functions for varint u32 and u64.